core/src/apps/nem/mosaic/helpers.py

```diff
@@ -6,7 +6,7 @@ def get_mosaic_definition(namespace_name: str, mosaic_name: str, network: int) -
         if namespace_name == m["namespace"] and mosaic_name == m["mosaic"]:
             if ("networks" not in m) or (network in m["networks"]):
                 return m
-    return None
+    return {}
```

Comment:
Changed to `None` in db28d00 - the usage site is using `if result`, so it would theoretically work with empty dict as well.

Also, I have defined a new `Mosaic` object that is returned instead of `dict` - but we could theoretically use already existing `NEMMosaicDefinition` message, I was not sure what is better

Reply:
Reusing `NEMMosaicDefinition` would work, the only problem are the optional fields. This depends on where this is called, and whether we can stick an `assert definition.name is not None` in these places **and it is valid**, iow, if the result is guaranteed to come from the predefined mosaic list and not from an outside use of `NEMMosaicDefinition`

also it might be a better choice here to move this function to `nem_mosaics.py` (generated from `.mako`), return the class instance directly from there, and convert the dict to an if/elif sequence (see `ethereum/networks.py`)

(the general reason to do this is that an if/elif sequence is stored in the flash memory, but the dict must be created in RAM)